core/src/main/java/com/linecorp/armeria/common/logging/RequestContextExporterBuilder.java

```diff
@@ -128,30 +143,54 @@ public boolean containsAttribute(AttributeKey<?> key) {
     @Deprecated
     public Map<String, AttributeKey<?>> getAttributes() {
         return attrs.stream().collect(
-                toImmutableMap(e -> e.exportKey.substring(PREFIX_ATTRS.length()), e -> e.key));
+                toImmutableMap(e -> {
+                    if (e.exportKey.startsWith(PREFIX_ATTRS)) {
+                        return e.exportKey.substring(PREFIX_ATTRS.length());
+                    }
+                    return e.exportKey;
+                }, e -> e.key));
     }
 
     /**
      * Adds the specified HTTP request header name to the export list.
      */
     public RequestContextExporterBuilder addHttpRequestHeader(CharSequence name) {
-        addHttpHeader(PREFIX_HTTP_REQ_HEADERS, httpReqHeaders, requireNonNull(name, "name"));
+        final AsciiString key = toHeaderName(requireNonNull(name, "name"));
+        final String exportKey = PREFIX_HTTP_REQ_HEADERS + key;
+        httpReqHeaders.add(new ExportEntry<>(key, exportKey));
+        return this;
```

Comment:
Ditto - Could delegate to `addHttpRequestHeader(CharSequence, String)`.

Reply:
Ditto because of the same reason (https://github.com/line/armeria/pull/2531#discussion_r385646413) :-)